tests/paramgrill.c

```diff
@@ -340,76 +891,264 @@ static int BMK_seed(winnerInfo_t* winners, const ZSTD_compressionParameters para
     return better;
 }
 
+/* bounds check in sanitize too? */
+#define CLAMP(var, lo, hi) {                          \
+    var = MAX(MIN(var, hi), lo);                      \
+}
 
 /* nullified useless params, to ensure count stats */
-static ZSTD_compressionParameters* sanitizeParams(ZSTD_compressionParameters params)
+/* no point in windowLog < chainLog (no point 2x chainLog for bt) */
+/* now with built in bounds-checking */
+/* no longer does anything with sanitizeVarArray + clampcheck */
+static ZSTD_compressionParameters sanitizeParams(ZSTD_compressionParameters params)
 {
-    g_params = params;
     if (params.strategy == ZSTD_fast)
         g_params.chainLog = 0, g_params.searchLog = 0;
```

Comment:
since you return `params` at the end of the function,
you should modify `params` instead of `g_params`.

Also : if `g_params` is no longer useful, you can remove it entirely.